easybuild/easyblocks/generic/mesonninja.py

```diff
@@ -0,0 +1,103 @@
+##
+# Copyright 2018-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing software with Meson & Ninja.
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.filetools import change_dir, mkdir
+from easybuild.tools.run import run_cmd
+
+
+class MesonNinja(EasyBlock):
+    """
+    Support for building and installing software with 'meson' and 'ninja'.
+    """
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Define extra easyconfig parameters specific to MesonNinja."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+        extra_vars.update({
+            'separate_build_dir': [True, "Perform build in a separate directory", CUSTOM],
+        })
+        return extra_vars
+
+    def configure_step(self, cmd_prefix=''):
+        """
+        Configure with Meson.
+        """
+        if self.cfg.get('separate_build_dir', True):
+            builddir = os.path.join(self.builddir, 'easybuild_obj')
+            mkdir(builddir)
+            change_dir(builddir)
+
+        cmd = "%(preconfigopts)s meson --prefix %(installdir)s %(configopts)s %(sourcedir)s" % {
+            'configopts': self.cfg['configopts'],
+            'installdir': self.installdir,
+            'preconfigopts': self.cfg['preconfigopts'],
+            'sourcedir': self.start_dir,
+        }
+        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+        return out
+
+    def build_step(self, verbose=False, path=None):
+        """
+        Build with Ninja.
+        """
+        parallel = ''
+        if self.cfg['parallel']:
+            parallel = "-j %s" % self.cfg['parallel']
```

Comment:
Aren't we doing -j by default? At least for configuremake and cmakemake?

Reply:
We are, `self.cfg['parallel']` is set to the available cores by default.

Reply:
oops, reading the code wrong (again)